### PR TITLE
Add usermgt audit logv2

### PR DESCRIPTION
**Pull request by VivekVinushanth**
### Proposed changes in this pull request

This PR on-boards audit loggers for the user management (users,  roles)

### CODE REVIEW

File: components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LogConstants.java [hunk 1]
```diff
@@ -58,4 +58,96 @@ public static class InputKeys {
...
+        public static final String SET_USER_CLAIM_VALUE_ACTION = "set-user-claim-value";
+        public static final String SET_USER_CLAIM_VALUES_ACTION = "set-user-claim-values";
+        public static final String DELETE_USER_CLAIM_VALUES_ACTION = "delete-user-claim-values";
+        public static final String DELETE_USER_CLAIM_VALUE_ACTION = "delete-user-claim-value";
+        public static final String CREDENTIAL_UPDATE_BY_USER_ACTION = "credential-update-by-user";
+        public static final String CREDENTIAL_UPDATE_BY_ADMIN_ACTION = "credential-update-by-administrator";
+        public static final String GET_USER_LIST_ACTION = "get-user-list";
+        public static final String GET_USER_CLAIM_VALUE_ACTION = "get-user-claim-value";
+        public static final String GET_USER_CLAIM_VALUES_ACTION = "get-user-claim-values";
+        public static final String UPDATE_USERS_OF_GROUP = "update-users-of-group";
```
Comment (kayathiri4) [inline components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LogConstants.java:94]:
```suggestion
        public static final String UPDATE_USERS_OF_GROUP_ACTION = "update-users-of-group";
```

File: components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LogConstants.java [hunk 2]
```diff
@@ -58,4 +58,96 @@ public static class InputKeys {
...
+        public static final String UPDATE_GROUPS_OF_USER_ACTION = "update-groups-of-user";
+
+        public static final String ADD_GROUP_ACTION = "add-group";
+        public static final String GET_GROUP_ACTION = "get-group";
+        public static final String DELETE_GROUP_ACTION = "delete-group";
+        public static final String UPDATE_GROUP_ACTION = "update-group";
+        public static final String UPDATE_GROUP_NAME_ACTION = "update-group-name";
+        public static final String GET_GROUP_LIST = "get-group-list";
```
Comment (kayathiri4) [inline components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LogConstants.java:102]:
```suggestion
        public static final String GET_GROUP_LIST_ACTION = "get-group-list";
```
Check below constants as well

File: components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java [hunk 3]
```diff
@@ -89,15 +114,15 @@ public static void triggerAuditLogEvent(AuditLog.AuditLogBuilder auditLogBuilder
...
-     * DiagnosticLog.DiagnosticLogBuilder)}.
      *
      * @param componentId    Component ID.
      * @param input          Input parameters.
      * @param resultStatus   Result status.
      * @param resultMessage  Result message.
      * @param actionId       Action ID.
      * @param configurations System/application level configurations.
+     * @Deprecated This method is deprecated. Use the method with {@link #triggerDiagnosticLogEvent(
+     *DiagnosticLog.DiagnosticLogBuilder)}.
```
Comment (kayathiri4) [inline components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java:125]:
```suggestion
     * DiagnosticLog.DiagnosticLogBuilder)}.
```

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+import static org.wso2.carbon.identity.role.v2.mgt.core.RoleConstants.ORGANIZATION;
+
+/**
+ * This v2 audit logger logs the RoleManagement related operations.
+ */
+public class RoleManagementV2AuditLogger extends AbstractRoleManagementListener {
+
+    @Override
+    public int getExecutionOrderId() {
+        return 0;
```
Comment (kayathiri4) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java:73]: Shall we add a new line after the method signatures.

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+
+    @Override
+    public void postUpdateRoleName(String roleId, String newRoleName, String tenantDomain) {
+
+        JSONObject data = new JSONObject();
+        data.put(ROLE_NAME_FIELD, newRoleName);
+        if (isEnable()) {
+            AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(
+                    ListenerUtils.getInitiatorId(), LoggerUtils.getInitiatorType(ListenerUtils.getInitiatorId()),
+                    roleId, LoggerUtils.Target.Role.name(),
```
Comment (kayathiri4) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java:132]:
```suggestion
                    roleId, LoggerUtils.Target.Role.name(),```

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java [hunk 6]
```diff
@@ -0,0 +1,315 @@
...
+
+
+    @Override
+    public void postDeleteRole(String roleId, String tenantDomain) {
+
+        if (isEnable()) {
+            AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(
+                    ListenerUtils.getInitiatorId(), LoggerUtils.getInitiatorType(ListenerUtils.getInitiatorId()),
+                    roleId, LoggerUtils.Target.Role.name(),
+                    DELETE_ROLE_ACTION);
```
Comment (kayathiri4) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java:146]: Do we need this in new line?

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/utils/ListenerUtils.java [hunk 7]
```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (kayathiri4) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/utils/ListenerUtils.java:2]: Can we update the license header?

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/utils/ListenerUtils.java [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+/**
+ * Utility class that handles the relevant utility tasks of listeners.
+ */
+public class ListenerUtils {
+
+
+    /**
+     * To get the current user, who is doing the current task.
+     *
+     * @return current logged-in user
```
Comment (kayathiri4) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/utils/ListenerUtils.java:39]:
```suggestion
     * @return current logged-in user.
```
Check other places as well

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/internal/UserMgtDSComponent.java [hunk 9]
```diff
@@ -32,17 +32,13 @@
...
 import org.wso2.carbon.user.core.service.RealmService;
 import org.wso2.carbon.user.mgt.RolePermissionManagementService;
 import org.wso2.carbon.user.mgt.RolePermissionManagementServiceImpl;
-import org.wso2.carbon.user.mgt.listeners.PermissionAuthorizationListener;
-import org.wso2.carbon.user.mgt.listeners.UserClaimsAuditLogger;
-import org.wso2.carbon.user.mgt.listeners.UserDeletionEventListener;
-import org.wso2.carbon.user.mgt.listeners.UserManagementAuditLogger;
-import org.wso2.carbon.user.mgt.listeners.UserMgtAuditLogger;
-import org.wso2.carbon.user.mgt.listeners.UserMgtFailureAuditLogger;
+import org.wso2.carbon.user.mgt.listeners.*;
```
Comment (kayathiri4) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/internal/UserMgtDSComponent.java:41]: Shall we revert this?

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/GroupManagementV2AuditLogger.java [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+            JSONObject dataObject = new JSONObject();
+            if (CollectionUtils.isNotEmpty(groupList)) {
+                dataObject.put(GROUPS_FIELD, new JSONArray(groupList));
+            }
+            AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(
+                    ListenerUtils.getInitiatorId(), LoggerUtils.getInitiatorType(ListenerUtils.getInitiatorId()),
+                    userId, Target.User.name(), GET_GROUPS_OF_USERS).data(jsonObjectToMap(dataObject));
+            triggerAuditLogEvent(auditLogBuilder, true);
+        }
+
```
Comment (kayathiri4) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/GroupManagementV2AuditLogger.java:134]:
```suggestion
```

File: components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java [hunk 11]
```diff
@@ -285,4 +310,40 @@ public static String getSanitizedErrorMessage(String errorMessage, String userNa
...
+    }
+
+    /**
+     * Get the Initiator type.
+     *
+     * @param initiator Initiator for the logs.
+     * @return Type of the initiator.
+     */
+    public static String getInitiatorType (String initiator) {
+        if (initiator.equals(LoggerUtils.Initiator.System.name())) {
```
Comment (VivekVinushanth) [inline components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LoggerUtils.java:344]: add a new line after the methods.

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java [hunk 12]
```diff
@@ -0,0 +1,302 @@
...
+                data.put(USERS_FIELD, new JSONArray(userList));
+            }
+            if (ArrayUtils.isNotEmpty(groupList.toArray())) {
+                data.put(GROUPS_FIELD, new JSONArray(groupList));
+            }
+            if (audience.equals(APPLICATION)) {
+                AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(
+                        ListenerUtils.getInitiatorId(), LoggerUtils.getInitiatorType(ListenerUtils.getInitiatorId()),
+                        roleId, LoggerUtils.Target.Role.name(),
+                        ADD_APP_ROLE_ACTION).data(jsonObjectToMap(data));
```
Comment (VivekVinushanth) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java:112]: setting the audience is enough.

Comment (VivekVinushanth) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java:112]: setting the audience is enough.

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java [hunk 13]
```diff
@@ -0,0 +1,302 @@
...
+            }
+            if (ArrayUtils.isNotEmpty(userList.toArray())) {
+                data.put(USERS_FIELD, new JSONArray(userList));
+            }
+            if (ArrayUtils.isNotEmpty(groupList.toArray())) {
+                data.put(GROUPS_FIELD, new JSONArray(groupList));
+            }
+            if (audience.equals(APPLICATION)) {
+                AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(
+                        ListenerUtils.getInitiatorId(), LoggerUtils.getInitiatorType(ListenerUtils.getInitiatorId()),
```
Comment (VivekVinushanth) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java:110]: Check if we need ListenerUtils.getInitiatorId();

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java [hunk 14]
```diff
@@ -0,0 +1,302 @@
...
+                        roleId, LoggerUtils.Target.Role.name(),
+                        ADD_APP_ROLE_ACTION).data(jsonObjectToMap(data));
+                triggerAuditLogEvent(auditLogBuilder, true);
+            }
+            if (audience.equals(ORGANIZATION)) {
+                AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(
+                        ListenerUtils.getInitiatorId(), LoggerUtils.getInitiatorType(ListenerUtils.getInitiatorId()),
+                        roleId, LoggerUtils.Target.Role.name(),
+                        ADD_ORG_ROLE_ACTION).data(jsonObjectToMap(data));
+                triggerAuditLogEvent(auditLogBuilder, true);
```
Comment (VivekVinushanth) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java:120]: overload the method if the possible.

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/utils/ListenerUtils.java [hunk 15]
```diff
@@ -0,0 +1,116 @@
...
+    }
+
+    /**
+     * This method will append the user store domain with user/role name.
+     *
+     * @param entity           Entity that need to modified.
+     * @param userStoreManager UserStore Manager particular user/role handled by.
+     * @return UserStoreDomain/UserName or UserStoreDomain/RoleName
+     */
+    public static String getEntityWithUserStoreDomain(String entity, UserStoreManager userStoreManager) {
```
Comment (VivekVinushanth) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/utils/ListenerUtils.java:61]: Check if we require this.

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/GroupManagementV2AuditLogger.java [hunk 16]
```diff
@@ -0,0 +1,146 @@
...
+        if (super.isEnable()) {
+            return isEnableV2AuditLogs();
+        }
+        return false;
+    }
+
+    public boolean postAddGroup(String groupName, String groupId, List<String> userIds, List<Claim> claims,
+                                UserStoreManager userStoreManager) {
+
+        if (isEnable()) {
```
Comment (VivekVinushanth) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/GroupManagementV2AuditLogger.java:63]: check if we require this as well.

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementV2AuditLogger.java [hunk 17]
```diff
@@ -0,0 +1,313 @@
...
+                                                       UserStoreManager userStoreManager) {
+
+        if (isEnable()) {
+            String initiatorId = ListenerUtils.getInitiatorId();
+            // In most cases even if the password reset is initiated by the user him self, it is being treated as
+            // admin initiated. Hence,
+            if (StringUtils.isNotBlank(initiatorId) && initiatorId.equals(userId)) {
+                AuditLog.AuditLogBuilder auditLogBuilder = new AuditLog.AuditLogBuilder(
+                        ListenerUtils.getInitiatorId(), LoggerUtils.getInitiatorType(ListenerUtils.getInitiatorId()),
+                        userId, LoggerUtils.Target.User.name(), CREDENTIAL_UPDATE_BY_ADMIN_ACTION);
```
Comment (VivekVinushanth) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementV2AuditLogger.java:187]: check if we require this and see if we can just generalize.

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementV2AuditLogger.java [hunk 18]
```diff
@@ -0,0 +1,313 @@
...
+                    ListenerUtils.getInitiatorId(), LoggerUtils.getInitiatorType(ListenerUtils.getInitiatorId()),
+                    userId, LoggerUtils.Target.User.name(), GET_USER_CLAIM_VALUES_ACTION)
+                    .data(jsonObjectToMap(dataObject));
+            triggerAuditLogEvent(auditLogBuilder, true);
+        }
+        return true;
+    }
+
+    @Override
+    public boolean doPostAuthenticateWithID(String userId, AuthenticationResult authenticationResult,
```
Comment (VivekVinushanth) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementV2AuditLogger.java:275]: Better remove it, if we are separating authentication to a new logger.

File: components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/utils/ListenerUtils.java [hunk 19]
```diff
@@ -65,14 +65,19 @@ public class ListenerUtils {
...
     public static final String CLAIM_VALUE_FIELD = "Claim Value";
     public static final String USERS_FIELD = "Users";
+    public static final String GROUPS_FIELD = "Groups";
+    public static final String GROUP_NAME_FIELD = "GroupName";
     public static final String PERMISSIONS_FIELD = "Permissions";
     public static final String DELETED_USERS = "Deleted Users";
     public static final String NEW_USERS = "New Users";
     public static final String DELETED_ROLES = "Deleted Roles";
     public static final String NEW_ROLES = "New Roles";
+    public static final String DELETED_GROUPS = "DeletedGroups";
```
Comment (VivekVinushanth) [inline components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/utils/ListenerUtils.java:75]: check if we need these as we have moved the constants to the logConstants.

File: components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LogConstants.java [hunk 20]
```diff
@@ -58,4 +58,98 @@ public static class InputKeys {
         public static final String ERROR_MESSAGE = "error message";
         public static final String CALLER_PATH = "common auth caller path";
     }
+
+    /**
+     * Application Mgt related log constants.
+     */
+    public static class ApplicationManagement {
```
Comment (piraveena) [inline components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LogConstants.java:65]: add new line

Comment (piraveena) [reply components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LogConstants.java:65]: check other places too

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java [hunk 21]
```diff
@@ -0,0 +1,233 @@
...
+                            List<Permission> permissions, String audience, String audienceId, String tenantDomain) {
+
+        JSONObject data = new JSONObject();
+        String roleId = roleBasicInfo.getId();
+        data.put(ROLE_NAME_FIELD, roleName);
+        data.put(AUDIENCE_FIELD, audience);
+        if (ArrayUtils.isNotEmpty(permissions.toArray())) {
+            ArrayList<String> permissionsList = new ArrayList<>();
+            for (Permission permission : permissions) {
+                permissionsList.add(permission.getName());
```
Comment (piraveena) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java:93]: we can use put and add all at once

Comment (piraveena) [reply components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java:93]: check other places too

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java [hunk 22]
```diff
@@ -0,0 +1,233 @@
...
+    }
+
+    @Override
+    public void postGetUserListOfRole(List<UserBasicInfo> userBasicInfoList, String roleId, String tenantDomain) {
+
+        JSONObject data = new JSONObject();
+        if (ArrayUtils.isNotEmpty(userBasicInfoList.toArray())) {
+            ArrayList<String> usersList = new ArrayList<>();
+            for (UserBasicInfo user : userBasicInfoList) {
+                usersList.add(user.getId());
```
Comment (piraveena) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java:136]: check previous comment

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java [hunk 23]
```diff
@@ -0,0 +1,233 @@
...
+                                         String tenantDomain) {
+
+        JSONObject data = new JSONObject();
+        if (ArrayUtils.isNotEmpty(newUserIDList.toArray())) {
+            data.put(NEW_USERS_FIELD, new JSONArray(newUserIDList));
+        }
+        if (ArrayUtils.isNotEmpty(deletedUserIDList.toArray())) {
+            data.put(DELETED_USERS_FIELD, new JSONArray(deletedUserIDList));
+        }
+        if (isEnable()) {
```
Comment (piraveena) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java:157]: do we need this

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/utils/ListenerUtils.java [hunk 24]
```diff
@@ -0,0 +1,98 @@
...
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
+
+/**
+ * Utility class that handles the relevant utility tasks of listeners.
+ */
+public class ListenerUtils {
+
+
```
Comment (piraveena) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/utils/ListenerUtils.java:35]: remove unnecessary line

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/utils/ListenerUtils.java [hunk 25]
```diff
@@ -0,0 +1,98 @@
...
+        }
+        return initiator;
+    }
+
+    /**
+     * Get the initiator for audit logs.
+     *
+     * @return Initiator id despite masking.
+     */
+    public static String getInitiatorId() {
```
Comment (piraveena) [inline components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/utils/ListenerUtils.java:82]: add new line

File: components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java (hunk 23 above)
Comment (VivekVinushanth) [reply components/role-mgt/org.wso2.carbon.identity.role.v2.mgt.core/src/main/java/org/wso2/carbon/identity/role/v2/mgt/core/listener/RoleManagementV2AuditLogger.java:157]: I checked that in certain cases the enabled checked is not respected. Hence, it is necessary to check if enabled at each event handling method. Hence, Will add 

```
if (!isEnable()) {
            return true;
        }
```

or 

```
if (!isEnable()) {
            return;
        }
```

at necessary places

**Review by kayathiri4 — APPROVED** [review]

File: components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LogConstants.java [hunk 26]
```diff
@@ -58,4 +58,101 @@ public static class InputKeys {
...
+        public static final String UPDATE_GROUP_NAME_ACTION = "update-group-name";
+        public static final String GET_GROUP_LIST_ACTION = "get-group-list";
+        public static final String GET_USERS_OF_GROUP_ACTION = "get-users-of-group";
+        public static final String GET_GROUPS_OF_USERS_ACTION = "get-groups-of-users";
+
+        public static final String DELETE_ROLE_ACTION = "delete-role";
+        public static final String ADD_ROLE_ACTION = "add-role";
+        public static final String GET_ROLE_LIST_ACTION = "get-role-list";
+        public static final String GET_ROLE_ACTION = "get-role";
+        public static final String GET_ROLE_BASIC_INFO_ACTION = "get-role-basicInformation";
```
Comment (piraveena) [inline components/central-logger/org.wso2.carbon.identity.central.log.mgt/src/main/java/org/wso2/carbon/identity/central/log/mgt/utils/LogConstants.java:112]: Shall we rename it? Other names are in simple case

**Review by piraveena — APPROVED** [review]

**Review by kayathiri4 — APPROVED** [review]

**Pull request merged**